coriolis/tests/scheduler/rpc/test_client.py

```diff
@@ -0,0 +1,241 @@
+            enabled=True)
+        self.assertEqual(result, service_mock1)
+
+    @mock.patch('random.choice')
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_random_choice(
+            self, get_workers_for_specs, mock_random_choice):
+        mock_service = mock.MagicMock()
+        get_workers_for_specs.return_value = [mock_service]
+
+        result = self.client.get_worker_service_for_specs(
+            mock.sentinel.ctxt, random_choice=False)
+
+        mock_random_choice.assert_not_called()
+        get_workers_for_specs.assert_called_once_with(
+            mock.sentinel.ctxt, provider_requirements=None, region_sets=None,
+            enabled=True)
+        self.assertEqual(result, mock_service)
+
+    @mock.patch.object(client.SchedulerClient, 'get_worker_service_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_different_platforms(
+            self, mock_get_task_runner_class, get_worker_service_for_specs):
+        for platform in [constants.TASK_PLATFORM_SOURCE,
+                         constants.TASK_PLATFORM_DESTINATION,
+                         constants.TASK_PLATFORM_BILATERAL]:
+            mock_get_task_runner_class.return_value.get_required_platform.\
+                return_value = platform
+            mock_get_task_runner_class.return_value.\
+                get_required_provider_types.return_value = {
+                    constants.PROVIDER_PLATFORM_SOURCE: 'provider_type'}
+            get_worker_service_for_specs.return_value = {'id': 'test_id'}
+
+            result = self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint, retry_period=0)
+
+            self.assertEqual(result, {'id': 'test_id'})
+            self.assertIsInstance(result, dict)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_retry(
+            self, mock_get_task_runner_class, mock_get_workers):
+        mock_get_task_runner_class.return_value.get_required_platform.\
+            return_value = constants.TASK_PLATFORM_SOURCE
+        mock_get_task_runner_class.return_value.get_required_provider_types.\
+            return_value = {constants.PROVIDER_PLATFORM_SOURCE:
+                            'provider_type'}
+        mock_get_workers.side_effect = [Exception(), [{'id': 'test_id'}]]
+
+        with self.assertLogs('coriolis.scheduler.rpc.client',
+                             level=logging.WARN):
+            self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint, retry_period=0)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
```

Comment:
Same: This should've mocked `get_worker_service_for_specs`

Reply:
This will also help with coverage, as currently `get_workers_for_specs` raises `NoSuitableWorkerServiceError`, instead of the tested method

Reply:
Gotcha, thank you!